src/core_modules/capture-core/components/WidgetProfile/hooks.js

```diff
@@ -1,7 +1,86 @@
 // @flow
+import { useState, useCallback, useEffect } from 'react';
+import { useDataEngine } from '@dhis2/app-runtime';
+import { makeQuerySingleResource } from 'capture-core/utils/api';
 import { useSelector } from 'react-redux';
+import type { apiProgramTrackedEntityAttribute } from 'capture-core/metaDataStoreLoaders/programs/quickStoreOperations/types';
+import { convertValue as convertServerToClient } from '../../converters/serverToClient';
+import { subValueGetterByElementType } from './getSubValueForTei';
 
 export const useProfileInfo = () => {
     const profileInformation = useSelector(({ enrollmentPage }) => enrollmentPage.profileInformation) ?? [];
     return { attributes: profileInformation };
 };
+
+type InputProgramData = {
+    programs: {
+        programTrackedEntityAttributes: Array<apiProgramTrackedEntityAttribute>
+    }
+}
+
+type InputAttribute = {
+    attribute: string,
+    code: string,
+    created: string,
+    displayName: string,
+    lastUpdated: string,
+    value: string,
+    valueType: string
+}
+
+type InputTEIData = {
+    trackedEntityInstances: {
+        attributes: Array<InputAttribute>
+    }
+}
+
+export const useListAttributes = (programsData: InputProgramData, trackedEntityInstancesData: InputTEIData) => {
+    const dataEngine = useDataEngine();
+
+    const [listAttributes, setListAttributes] = useState([]);
+
+    const getListAttributes = useCallback(async () => {
+        if (programsData && trackedEntityInstancesData) {
+            const querySingleResource = makeQuerySingleResource(dataEngine.query.bind(dataEngine));
+
+            const { programs: { programTrackedEntityAttributes } } = programsData;
+            const { trackedEntityInstances: { attributes } } = trackedEntityInstancesData;
+            const computedAttributes = await programTrackedEntityAttributes
+                .filter(item => item.displayInList)
```

Comment:
We should remove the filtering here (if I'm not missing something)

Reply:
You can absolutely make a case for keeping this filtering and letting the edit mode retrieve the values that are not displayInList. But I suggest we do it like this for now, we can improve later if needed.